### PR TITLE
Azure devops pipeline as a code

### PR DESCRIPTION
**Pull request by LassiAutioSolita**
My first blog post in Solita.

I first thought to write about CI and CD but thought maybe I'll write now only about CI and maybe write another blog post about CD. Thoughts about this?

Is this too simple or detailed? What should I include or exclude?

### CODE REVIEW

File: _posts/2020-11-xx-azure-devops-pipeline-as-code.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+excerpt: >
+  TODO: put an excerpt here
+tags:
+ - Azure DevOps
+ - DevOps
+ - continuous delivery
+ - continuous integration
+---
+
+Azure DevOps used to have only graphical build pipelines (known as classic pipelines) which were edited only in the web portal. In Build 2019 Microsoft released pipeline as code were pipelines are defined with YAML. In this blog post I will go through the basics of YAML pipelines and compare them with the older classic pipelines.
```
Comment (LassiAutioSolita) [inline _posts/2020-11-xx-azure-devops-pipeline-as-code.md:14]: Intro is yet draft.

**Review by solita-jrutila — COMMENTED** [review]
Didn't spot any grammar or other problems with a quick glance.

File: _posts/2020-11-xx-azure-devops-pipeline-as-code.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+ - Azure DevOps
+ - DevOps
+ - continuous integration
+ - Build pipeline
+ - CI
+---
+
+Azure DevOps used to have only graphical build pipelines, known as Classic Pipeline GUI, which were edited only in the web portal.
+In Build 2019 Microsoft released pipeline as code, where pipelines are defined with YAML (aka YAML pipeline).
+In this blog post, I will go through the basics of YAML pipelines and compare them with the classic pipelines.
```
Comment (crystoll) [inline _posts/2020-11-xx-azure-devops-pipeline-as-code.md:19]: This is the part where we typically try to "sell" the blog: Make readers interested in investing a bit of time to read it. This tells what the blog is about, which is good, but would it be possible to find more impact by telling what could be achieved by it? Even one sentence explaining why pipelines as code are cool, or what is the effect if you start using this?

File: _posts/2020-11-xx-azure-devops-pipeline-as-code.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+- Pipelines for continuous integration, delivery, and deployment,
+- Test Plans, to run tests and
+- Artifacts to save various kinds of packages (NuGet, NPM, etc.).
+
+Practically it includes most of the things a software team needs except infrastructure to where to run the software.
+Its strength is the smooth integration with all of these features. For example, we can access source code repo from build pipelines, and run build automatically when there is a new commit. Or we can mark tasks on the board as done when its code is deployed via release pipeline.
+
+Even if Azure DevOps is Microsoft's product, it can be used with not-Microsoft programming languages and deploy to any AWS or GCP along with Azure or on-premises. You can use it even if you're not using Microsoft's products.
+
+In this blog post, we will concentrate on pipelines; as code!
```
Comment (crystoll) [inline _posts/2020-11-xx-azure-devops-pipeline-as-code.md:36]: This is a good introduction, giving a bit of an introduction and overview on things, to get everyone on same line in concise manner.

File: _posts/2020-11-xx-azure-devops-pipeline-as-code.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+Some of us might doubt if it is too difficult to start using YAML pipelines.
+Fortunately, there isn't a need to know everything by heart.
+There is some graphical help in the Azure DevOps web portal and IntelliSense will also help with the web portal and Visual Studio Code.
+If you are in doubt about YAML pipelines, I recommend to give them a chance.
+They can be the next upgrade for your pipelines and efficiency.
+
+## Further Reading
+
+* [Use Azure Pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/get-started/pipelines-get-started?view=azure-devops)
+* [YAML schema reference](https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema)
```
Comment (crystoll) [inline _posts/2020-11-xx-azure-devops-pipeline-as-code.md:234]: Good links. Possibly include a link to another Solita blog that deals with Azure devops? It's not the same viewpoint, but nice extra story on the theme. Cross-linking each others blogs makes them always much more visible. I found this with google search: "solita infrastructure as code" :)

https://data.solita.fi/automatized-code-deployment-from-azure-devops-to-databricks/

Comment (LassiAutioSolita) [reply _posts/2020-11-xx-azure-devops-pipeline-as-code.md]: Thanks! I added that link. I searched only from dev blog and didn't realize we have other blogs also. Good reminder for me.

**Pull request merged**